client/my-sites/checkout/composite-checkout/components/prepurchase-notices/index.jsx

```diff
@@ -93,28 +95,40 @@ const PrePurchaseNotices = () => {
 		return matchingProducts?.[ 0 ];
 	}, [ currentSiteProducts, cartItemSlugs ] );
 
-	/**
-	 * The product currently in the cart that overlaps/conflicts with the current active site plan.
-	 */
-	const cartProductThatOverlapsSitePlan = useSelector( ( state ) => {
-		const backupSlugInCart = cartItemSlugs.find( isJetpackBackupSlug );
-		const antiSpamSlugInCart = cartItemSlugs.find( isJetpackAntiSpamSlug );
-		const scanSlugInCart = cartItemSlugs.find( isJetpackScanSlug );
+	const cartProductThatOverlapsSitePlan = useMemo( () => {
+		const planSlugOnSite = currentSitePlan?.product_slug;
 
-		if ( backupSlugInCart && siteHasFeature( state, siteId, WPCOM_FEATURES_BACKUPS ) ) {
-			return getProductFromSlug( backupSlugInCart );
+		if ( ! planSlugOnSite || planSlugOnSite === 'jetpack_free' ) {
+			return null;
 		}
 
-		if ( antiSpamSlugInCart && siteHasFeature( state, siteId, WPCOM_FEATURES_ANTISPAM ) ) {
-			return getProductFromSlug( antiSpamSlugInCart );
-		}
+		const getMatchingProducts = ( cartItems, planSlug ) => {
+			// Get all features for the site plan
+			const planFeatures = getAllFeaturesForPlan( planSlug );
 
-		if ( scanSlugInCart && siteHasFeature( state, siteId, WPCOM_FEATURES_SCAN ) ) {
-			return getProductFromSlug( scanSlugInCart );
-		}
+			// Filter the cart items to only include those in the plan items or are inferior features to the plan feature
+			const matchingProducts = cartItems.filter( ( cartItem ) => {
```

Comment:
Would it make sense to extract `matchingProducts` here, since an almost similar version is used above?